src/webots/gui/WbMainWindow.cpp

```diff
@@ -2221,14 +2246,13 @@ void WbMainWindow::setWorldLoadingStatus(const QString &status) {
 
 void WbMainWindow::startAnimationRecording() {
   WbSimulationState::Mode currentMode = WbSimulationState::instance()->mode();
-  WbSimulationState::instance()->setMode(WbSimulationState::PAUSE);
-  const QString &worldFileName = WbWorld::instance()->fileName();
+  QString fileName = findHtmlFileName("Save Animation File");
```

Comment:
It does not work because the `WbProject::computeBestPathForSaveAs` does not take a const. (I think that is also why the other `QString fileName` of `WbMainWindow` are not const either.